### PR TITLE
[RW-9518][risk=no] [tests] Cromwell / User App tests

### PR DESCRIPTION
**Pull request by NehaBroad**
Future work to mark the story as done:
1) e2e for R

Improvements:
1) Correct snippet to use cromshell-alpha not cromshell_alpha 
3) Should we clean up the cromshell jobs before deleting cromwell? 
4) Having Notebook upload multiple files rather than one at a time
 

Description:

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [x] I have added explanatory comments where the logic is not obvious
- [x] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

**Review by NehaBroad — COMMENTED** [review]
Not sure if we need the other test Create and delete a Cromwell GKE app..thoughts?

File: e2e/app/page/notebook-page.ts [hunk 1]
```diff
@@ -713,4 +713,25 @@ export default class NotebookPage extends NotebookFrame {
     await this.page.keyboard.press('Enter', { delay: 20 });
     await this.page.keyboard.up(keyboardCommand);
   }
+
+  async selectSnippet(snippetsName: string, subCategory: string): Promise<void> {
```
[the remark this answers is not in the document]
Comment (jmthibault79) [reply e2e/app/page/notebook-page.ts:717]: Thanks for figuring this out!

**Review by jmthibault79 — CHANGES_REQUESTED** [review]

File: e2e/tests/nightly/gke-apps/cromwell.spec.ts [hunk 2]
```diff
@@ -66,29 +76,71 @@ describe('Cromwell GKE App', () => {
...
-    const deleteButton = new Button(page, deleteXPath);
-    expect(await deleteButton.exists()).toBeTruthy();
-    await deleteButton.click();
+    const isDeleted = cromwellPanel.deleteCromwellGkeApp();
+    expect(isDeleted).toBeTruthy();
+  });
+
+  test('Run cromwell using python notebook', async () => {
+    // Create or re-use workspace
+    await findOrCreateWorkspace(page, { workspaceName });
```
Comment (jmthibault79) [inline e2e/tests/nightly/gke-apps/cromwell.spec.ts:85]: we should use a separate workspace for this, so they don't interfere with each other

File: e2e/tests/nightly/gke-apps/cromwell.spec.ts [hunk 3]
```diff
@@ -66,29 +76,71 @@ describe('Cromwell GKE App', () => {
 
     await appsPanel.pollForStatus(expandedCromwellXpath, 'Running', 15 * 60e3);
 
-    const deleteXPath = `${expandedCromwellXpath}//*[@data-test-id="Cromwell-delete-button"]`;
-    const deleteButton = new Button(page, deleteXPath);
-    expect(await deleteButton.exists()).toBeTruthy();
-    await deleteButton.click();
+    const isDeleted = cromwellPanel.deleteCromwellGkeApp();
```
[the remark this answers is not in the document]
Comment (NehaBroad) [reply e2e/tests/nightly/gke-apps/cromwell.spec.ts]: Done

File: e2e/tests/nightly/gke-apps/cromwell.spec.ts [hunk 4]
```diff
@@ -4,10 +4,19 @@ import Button from 'app/element/button';
 import CromwellConfigurationPanel from 'app/sidebar/cromwell-configuration-panel';
 import BaseElement from 'app/element/base-element';
 import { waitForFn } from 'utils/waits-utils';
-import WarningDeleteCromwellModal from 'app/modal/warning-delete-cromwell-modal';
+import WorkspaceDataPage from 'app/page/workspace-data-page';
+import { makeRandomName } from 'utils/str-utils';
+import { Language } from 'app/text-labels';
+import expect from 'expect';
```
[the remark this answers is not in the document]
Comment (NehaBroad) [reply e2e/tests/nightly/gke-apps/cromwell.spec.ts]: removed expect

File: e2e/resources/cromwell/cromwell.wdl [hunk 5]
```diff
@@ -0,0 +1,20 @@
+version 1.0
```
[the remark this answers is not in the document]
Comment (NehaBroad) [reply e2e/resources/cromwell/cromwell.wdl]: updated wdl

File: e2e/resources/cromwell/cromwell.json [hunk 6]
```diff
@@ -0,0 +1 @@
+{}
```
[the remark this answers is not in the document]
Comment (NehaBroad) [reply e2e/resources/cromwell/cromwell.json]: done: changed to empty.json

File: e2e/app/sidebar/cromwell-configuration-panel.ts [hunk 7]
```diff
@@ -1,11 +1,71 @@
...
+    const createXPath = `${configPanel.getXpath()}//*[@id="Cromwell-cloud-environment-create-button"]`;
+    const createButton = new Button(page, createXPath);
+    expect(await createButton.exists()).toBeTruthy();
+    await createButton.click();
+
+    await appsPanel.close();
+    return;
+  }
+
+  async deleteCromwellGkeApp(): Promise<boolean> {
```
[the remark this answers is not in the document]
Comment (NehaBroad) [reply e2e/app/sidebar/cromwell-configuration-panel.ts]: Moved it to apps-panel

File: e2e/app/sidebar/cromwell-configuration-panel.ts [hunk 8]
```diff
@@ -1,11 +1,71 @@
 import { Page } from 'puppeteer';
 import { SideBarLink } from 'app/text-labels';
 import BaseEnvironmentPanel from './base-environment-panel';
+import Button from '../element/button';
+import expect from 'expect';
```
[the remark this answers is not in the document]
Comment (NehaBroad) [reply e2e/app/sidebar/cromwell-configuration-panel.ts]: Done

File: e2e/app/sidebar/apps-panel.ts [hunk 9]
```diff
@@ -3,6 +3,7 @@ import { SideBarLink } from 'app/text-labels';
 import BaseEnvironmentPanel from './base-environment-panel';
 import { waitForFn } from 'utils/waits-utils';
 import BaseElement from 'app/element/base-element';
+import expect from 'expect';
```
[the remark this answers is not in the document]
Comment (NehaBroad) [reply e2e/app/sidebar/apps-panel.ts:7]: done

[the remark this answers is not in the document]
Comment (jmthibault79) [reply e2e/app/sidebar/apps-panel.ts:7]: this is not done.

**Review by jmthibault79 — APPROVED** [review]

**Pull request merged**